tests/test_is_valid_email.py

```diff
@@ -1,121 +1,67 @@
-    Such an email should not be considered valid.
-    """
-    email = "test.user!#%^&*()+=[]{}|;<>?@inspection.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_email_with_lookalike_utf8_characters():
-    """
-    Test the is_valid_email function with an email containing look-alike UTF-8 characters.
-    Such an email should not be considered valid as it's potentially misleading.
-    """
-    # Using Cyrillic 'а' which looks like Latin 'a'
-    email = "test.userа@inspection.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_misspelled_inspection_subdomain():
-    """
-    Test the is_valid_email function with a misspelled "inspection" subdomain.
-    """
-    email = "user@inspektion.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_extra_characters_in_canada_domain():
-    """
-    Test the is_valid_email function with an extra character in the "canada" domain.
-    """
-    email = "user@canadaa.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_missing_domain_extension():
-    """
-    Test the is_valid_email function with a missing domain extension.
-    """
-    email = "user@canada."
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_missing_subdomain():
-    """
-    Test the is_valid_email function with a missing subdomain.
-    """
-    email = "user@.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_extra_characters_in_tld():
-    """
-    Test the is_valid_email function with extra characters in the TLD.
-    """
-    email = "user@inspection.gc.caa"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
+    def test_invalid_email_from_disallowed_domain(self):
+        self.assert_invalid_email(
+            "test.user@notallowed.com", self.email_config.validation_pattern
+        )
+
+    def test_concatenated_emails(self):
+        self.assert_invalid_email(
+            "test.user1@inspection.gc.catest.user2@inspection.gc.ca",
+            self.email_config.validation_pattern,
+        )
+
+    def test_email_with_unsupported_symbols(self):
+        self.assert_invalid_email(
+            "test.user!#%^&*()+=[]{}|;<>?@inspection.gc.ca",
```

Comment:
the problem with this test is that we want to fail on ANY of the invalid characters being present, but we test with an example that has ALL the invalid characters. we should loop and test for emails with each individual invalid characters.